### PR TITLE
rgw: remove metadata entry when remote fetch reply with ENOENT

### PR DESCRIPTION
**Pull request by yuvalif**
Signed-off-by: yuval Lifshitz <ylifshit@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

File: src/rgw/rgw_sync.cc [hunk 1]
```diff
@@ -1322,13 +1322,14 @@ int RGWMetaSyncSingleEntryCR::operate(const DoutPrefixProvider *dpp) {
       sync_status = retcode;
 
       if (sync_status == -ENOENT) {
-        /* FIXME: do we need to remove the entry from the local zone? */
+        tn->log(10, SSTR("removing local metadata entry:" << section << ":" << key));
+        yield call(new RGWMetaRemoveEntryCR(sync_env, raw_key));
```
Comment (cbodley) [inline src/rgw/rgw_sync.cc:1326]: what does this fix? i don't see any explanation in the commit message

doesn't this just remove retries from `RGWMetaRemoveEntryCR`, even though it still runs through that retry loop below?

these `Store`/`Remove` calls use `RGWObjVersionTracker` to detect races, so i think it's important to retry on `ECANCELED`

Comment (yuvalif) [reply src/rgw/rgw_sync.cc:1326]: you are right. the correct fix is to treat `ENOENT` on `RGWMetaRemoveEntryCR` as success

**Comment by yuvalif** [comment]
teuthology is passing **with valgrind issue**:
http://qa-proxy.ceph.com/teuthology/yuvalif-2022-08-08_07:48:26-rgw:verify-wip-yuval-test1-distro-default-smithi/6960935/teuthology.log

**Review by cbodley — APPROVED** [review]

**Comment by yuvalif** [comment]
> @yuvalif the linked results were from rgw/verify with a s3select test failure. has this been through rgw/multisite?

multisite runs fail in a way that is not useful. e.g.
http://pulpito.front.sepia.ceph.com/yuvalif-2022-08-07_12:59:16-rgw:multisite-wip-yuval-test1-distro-default-smithi/

**Comment by yuvalif** [comment]
teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2022-08-14_11:49:37-rgw-wip-yuval-test-aug-14-distro-default-smithi/

* valgrind issues
* "Cannot create secret" failure
* LC failures (see: a-proxy.ceph.com/teuthology/yuvalif-2022-08-14_11:49:37-rgw-wip-yuval-test-aug-14-distro-default-smithi/6972232/teuthology.log):
  * `s3tests_boto3.functional.test_s3.test_lifecycle_cloud_transition`
  * `s3tests_boto3.functional.test_s3.test_lifecycle_cloud_transition_large_obj`
* many multisite tests are failing (see: http://qa-proxy.ceph.com/teuthology/yuvalif-2022-08-14_11:49:37-rgw-wip-yuval-test-aug-14-distro-default-smithi/6972238/teuthology.log) on: `AssertionError: failed meta checkpoint for zone=test-zone2`
* tempest tests are failing: http://qa-proxy.ceph.com/teuthology/yuvalif-2022-08-14_11:49:37-rgw-wip-yuval-test-aug-14-distro-default-smithi/6972242/teuthology.log
* s3select failures: `s3tests_boto3.functional.test_s3select.test_progress_expressions` in: http://qa-proxy.ceph.com/teuthology/yuvalif-2022-08-14_11:49:37-rgw-wip-yuval-test-aug-14-distro-default-smithi/6972278/teuthology.log

**Pull request merged**